src/symbols.h

```diff
@@ -8,13 +8,24 @@
 
 #include "codeCache.h"
 #include "mutex.h"
+#include <dlfcn.h>
 
 
 class Symbols {
   private:
     static Mutex _parse_lock;
     static bool _have_kernel_symbols;
     static bool _libs_limit_reported;
+    static const void* _main_phdr;
+    static const char* _ld_base;
+
+    static bool isMainExecutable(const char* image_base, const void* map_end) {
+      return _main_phdr != NULL && _main_phdr >= image_base && _main_phdr < map_end;
+    }
+
+    static bool isLoader(const char* image_base) {
+      return image_base != NULL && _ld_base == image_base;
```

Comment:
I think the `!= NULL` check should be against _ld_base

```suggestion
      return _ld_base != NULL && _ld_base == image_base;
```